service/src/test/java/bio/terra/landingzone/library/landingzones/management/ResourcesDeleteManagerTest.java

```diff
@@ -0,0 +1,171 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.deployment.LandingZoneTagKeys;
+import bio.terra.landingzone.library.landingzones.management.deleterules.LandingZoneRuleDeleteException;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.management.exception.ManagementError;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.network.models.PrivateEndpoint;
+import com.azure.resourcemanager.network.models.PrivateEndpoints;
+import com.azure.resourcemanager.resources.models.GenericResource;
+import com.azure.resourcemanager.resources.models.GenericResources;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+
+@Tag("unit")
+@ExtendWith(MockitoExtension.class)
+class ResourcesDeleteManagerTest {
+  private static final String LANDING_ZONE_ID = UUID.randomUUID().toString();
+  private static final String RESOURCE_GROUP_NAME = "RESOURCE_GROUP";
+
+  @Mock private ArmManagers armManagersMock;
+  @Mock private DeleteRulesVerifier deleteRulesVerifierMock;
+
+  @Mock private AzureResourceManager azureResourceManagerMock;
+  @Mock private PrivateEndpoints privateEndpointsManagerMock;
+  @Mock private GenericResources genericResourcesManagerMock;
+
+  private ResourcesDeleteManager resourcesDeleteManager;
+
+  @BeforeEach
+  void setup() {
+    mockArmManager();
+    resourcesDeleteManager = new ResourcesDeleteManager(armManagersMock, deleteRulesVerifierMock);
+  }
+
+  @Test
+  void delete_LZResourcesNotFound() throws LandingZoneRuleDeleteException {
+    mockResourceListing(Collections.emptyList(), Collections.emptyList());
+
+    resourcesDeleteManager.deleteLandingZoneResources(LANDING_ZONE_ID, RESOURCE_GROUP_NAME);
+
+    verify(deleteRulesVerifierMock, times(1)).checkIfRulesAllowDelete(any());
+    verify(genericResourcesManagerMock, never()).deleteById(any());
+  }
+
+  @Test
+  void delete_SomeLZResourcesExist() throws LandingZoneRuleDeleteException {
+    var genericResource = mock(GenericResource.class);
+    when(genericResource.tags())
+        .thenReturn(Map.of(LandingZoneTagKeys.LANDING_ZONE_ID.toString(), LANDING_ZONE_ID));
+    mockResourceListing(
+        Collections.emptyList() /*ignoring private endpoints*/,
+        Collections.singletonList(genericResource));
+
+    resourcesDeleteManager.deleteLandingZoneResources(LANDING_ZONE_ID, RESOURCE_GROUP_NAME);
```

Comment:
Is there anything useful to be asserted about the return value from this method?

Reply:
I think the number of resources processed and the fact that they correspond to the provided resources. Added assertion.